docs/dev/adding_instruments.rst

```diff
@@ -812,6 +819,122 @@ In the above example, :code:`MultimeterA` and :code:`MultimeterB` use a differen
+
+    class InstrumentWithChannels(Instrument):
+        """An instrument with a channel."""
+        channels = Instrument.children(("A", "B"), VoltageChannel)
+
+        def __init__(self, adapter):
+            super().__init__(adapter, "Instrument with Channels")
+
+.. testcode:: with-protocol-tests
+    :hide:
+
+    with expected_protocol(InstrumentWithChannels,
+        [("SOURceA:VOLT 1.23", None), ("SOURceB:VOLT?", "1.23")]
+    ) as inst:
+        inst.ch_A.voltage = 1.23
+        assert inst.ch_B.voltage == 1.23
+
+If you set the voltage of the first channel of above :class:`ExtremeChannel` instrument with :code:`inst.chA.voltage = 1.23`, the driver sends :code:`"SOURceA:VOLT 1.23"` to the device, supplying the "A" of the channel name. The same channel could be called with :code:`inst.channels[0].voltage = 1.23` as well.
+
+
+Channels with fix prefix
+************************
+
+If all channel communication is prefixed by a specific command, e.g. :code:`"SOURceA:"` for channel A, you can subclass the channels :meth:`insert_id` method.
+
+.. testcode:: with-protocol-tests
+
+    class VoltageChannelPrefix(Channel):
+        """A channel of a voltage source, every command has the same prefix."""
+
+        def insert_id(self, command):
+            return f"SOURce{self.id}:{command}"
+
+        voltage = Channel.control(
+            "VOLT?", "VOLT %g",
+            """Control the output voltage of this channel.""",
+        )
+
+.. testcode:: with-protocol-tests
+    :hide:
+
+    class InstrumentWithChannelsPrefix(Instrument):
+        """An instrument with a channel."""
+
+        def __init__(self, adapter):
+            super().__init__(adapter, "Instrument with Channels")
+            self.add_child(VoltageChannelPrefix, "A")
+
+    with expected_protocol(InstrumentWithChannelsPrefix,
+        [("SOURceA:VOLT 1.23", None), ("SOURceA:VOLT?", "1.23")]
+    ) as inst:
+        inst.ch_A.voltage = 1.23
+        assert inst.ch_A.voltage == 1.23
+
+This channel class implements the same communication as the previous example, but implements the channel prefix in the :meth:`insert_id` method and not in the individual property (created by :meth:`control`).
+
+
+Collections of different channel types
+**************************************
+
+Some devices have different types of channels. In this case, you can specify a different `collection` and `prefix` parameter.
+
+.. testcode:: with-protocol-tests
+
+    class PowerChannel(Channel):
+        """A channel controlling the power."""
+
+        power = Channel.measurement(
+            "POWER?", """Measure the currently consumed power.""")
+
+    class MultiChannelTypeInstrument(Instrument):
+        """An instrument with two different channel types."""
+        analog = Instrument.children(("A", "B"), (VoltageChannel, VoltageChannelPrefix), prefix="an")
+        digital = Instrument.children((0, 1, 2), VoltageChannel, prefix="di")
+        power = Instrument.children(None, PowerChannel)
```

Comment:
On second thought -- what does the `id` of `instrument.power` become, here? and what is the placeholder that the command strings need? Needs to be included in the doctest below, I think.

Reply:
Good point. I was focused on the children and forgot the channels.

We use the ID to distinguish different channels of the same type.
If there is only a single "channel", you can hardcode the command string. This feature seems useful for different parts of a device (e.g. distinguish between the voltage source and a voltage measurement) and not if you just have one channel while there could be more (e.g. a one voltage source).

I'm thinking about a better solution. 

I see the point, that it would be good to have some ID as well

Reply:
Code changed, see other comment in code (now "prefix" toggles the modes and id is always possible).